File: IAIso-v5.0/core/iaiso-php/src/Policy/Aggregator.php
<?php

declare(strict_types=1);

namespace IAIso\Policy;

/** A coordinator aggregation strategy. */
interface Aggregator
{
    /** Wire-format aggregator name. */
    public function name(): string;

    /**
     * Compute the aggregate from per-execution pressures.
     *
     * @param array<string,float> $pressures
     */
    public function aggregate(array $pressures): float;
}

File: IAIso-v5.0/core/iaiso-java/iaiso-policy/src/main/java/io/iaiso/policy/Aggregator.java
package io.iaiso.policy;

import java.util.Map;

/** A coordinator aggregation strategy. */
public interface Aggregator {

    /** Wire-format aggregator name. */
    String name();

    /** Compute the aggregate from per-execution pressures. */
    double aggregate(Map<String, Double> pressures);
}
